upgrade to kaleido 1.0

bikegeek opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

    testing with Command line and IDE 
    verified GHA unit tests are passing

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
 
verify unit tests are passing

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[NA]**

- [x] Do these changes include sufficient testing updates? **[NA]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Do these changes introduce new SonarQube findings? **[Maybe]**</br>
If **yes**, please describe:

May have code coverage results below threshold

- [ ] Please complete this pull request review by **[June 30]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **METplotpy-X.Y.Z Development** project for official releases
- [ ] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

jprestop committed:
Feature METplus2857 main v3.1 support (#512)

* Making index consistent with other METplus repositories

* Initial commit of overview

georgemccabe committed:
Per #515, remove deprecated import (#516)

bikegeek committed:
Update base_plot.py

comment out the pio.kaleido.scope line, this is now causing tests to fail due to kaleido warning for deprecations in September 2025

bikegeek committed:
Update base_plot.py

backing out the latest from pio.kaleido line.  WARNING: GHA tests may fail due to upcoming deprecation and warnings that emanate due to Python 3.12 still features that will be deprecated.

bikegeek committed:
Update test_scatter.py

Tests pass outside of GHA.  Something in GHA is causing issues with Python 3.12 and kaleido (causes AssertionError in kaleido)

bikegeek committed:
upgrade to kaleido 1.0

jprestop reviewed (dismissed):
I reviewed the change to the one file and verified that all tests pass. I approve this request. Thank you for all of your work on this task, @bikegeek! It's great to see kaleido updated to a 1.0.0 version!

jprestop committed:
Adding modulefile for new machine ursa

bikegeek committed:
Feature 525 update plotly kaleido (#526)

* Update conf.py

update version for 3.20 beta1 development

* Feature 504 installation files (#511)

* Update Orion file for Python 3.12

* Per #504, updating Casper file for Python 3.12

* Per # 504, updating compiler for Orion

* Per #504, updating hera file for Python 3.12

* Per #504, updating jet file for Python 3.12

* Feature METplus2857 support (#513)

* Making index consistent with other METplus repositories

* Initial commit of overview

* Bugfix #515 develop - remove deprecated import (#517)

* Per #515, remove deprecated import

* remove additional unused import to resolve SQ

* Update base_plot.py

Comment out pio.kaleido.scope.chromium line.  Python 3.12 updates with updated deprecations in kaleido will cause some tests to fail due to these warnings.

* Remove soon to be deprecated call to plotly.io.kaleido.scope

* Comment out soon-to-be deprecated call to plolty.io.kaleido.scope

* add call to install chrome

* Add test_* to trigger and setup test to use test_requirements.txt

* To test plotly/kaleido updates

* Try using kaleido_get_chrome_sync

* Update import to use ploty.io._kaleido

* Using latest import, invoke the get_chrome() method instead of get_chrome_sync()

* explicitly pip install kaleido 1.0

* Add Google Chrome action

* fix syntax error

* remove - from the run command

* Explicitly get the Chrome environment, bypass the calls to plotly.io.kaleido for installing chrome, which already exists

* Set up a CHROME_INSTALL path and save it as an environment

* remove no-sudo

* removed extra line at top of file

* move env definition by indenting

* Removed all imports that didn't work and the soon to be deprecated reference to kaleido.scope.  Running after installing Chrome without setting an ENV variable.

* Remove any attempt to set an env

* Remove the --version to the path under the chrome installation

* Remove skip

* Update test_requirements.txt

match up with requirements.txt except for kaleido and plotly

* Attempt to get chrome via kaleido.get_chrome_sync()

* Update unit_tests.yaml

Comment out installation of Chrome for now but don't delete it in the event that Chrome will need to be explicitly available.

* Update base_plot.py

Remove old comment and update to reflect upcoming changes to kaleido

* Replace test_requirements.txt with requirements.txt

* Update Plotly from 6.0.0 to 6.1.1 and kaleido from 0.2.1 to 1.0.0

* Adding modulefile for new machine ursa

---------

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>
Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

bikegeek committed:
Merge branch 'main_v3.1' into feature_kaleido
